src/sentry/models/useroption.py

```diff
@@ -44,9 +35,28 @@ def get_value(self, user, key, default=None, **kwargs):
         return result.get(key, default)
 
     def unset_value(self, user, project, key):
-        # this isn't implemented for user-organization scoped options yet, because
-        # it hasn't been needed
-        self.filter(user=user, project=project, key=key).delete()
+        """
+        This isn't implemented for user-organization scoped options yet, because it hasn't been needed.
+        """
+        from sentry.models.notificationsetting import NotificationSetting
+
+        with transaction.atomic():
+            if key in [
+                "workflow:notifications",
+                "mail:alert",
+                "deploy-emails",
+                "subscribe_by_default",
+            ]:
+                type = get_key_from_legacy(key)
+                NotificationSetting.objects.remove_settings(
+                    ExternalProviders.EMAIL,
+                    type,
+                    user=user,
+                    project=project,
+                )
+
+                self.filter(user=user, project=project, key=key).delete()
```

Comment:
is this not used for other types of user settings?

Reply:
Good catch I'll fix that now.